internal/clients/composer/client.go

```diff
@@ -124,8 +123,8 @@ func (cc *ComposerClient) request(method, url string, headers map[string]string,
 		if err != nil {
 			return nil, err
 		}
-
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token()))
+		token = cc.Tokener.Token(req.Context()) // Use the Token method from the Tokener
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 		resp, err = cc.client.Do(req)
 	}
```

Comment:
The whole purpose of this refactoring is to get rid of the double request. Remove this, a client request becomes a single request, you will need to update many tests and that is expected.